src/common/config.cc

```diff
@@ -393,6 +393,54 @@ void md_config_t::show_config(Formatter *f)
   _show_config(NULL, f);
 }
 
+void md_config_t::config_options(Formatter *f)
+{
+  Mutex::Locker l(lock);
+  f->open_array_section("options");
+  for (const auto& i: schema) {
+    const Option &opt = i.second;
+    f->open_object_section("option");
```

Comment:
The `Option` class has a `dump` method that you can use instead of dumping all fields explicitly here.